tests/sys_architecture/Makefile

```diff
@@ -0,0 +1,18 @@
+BOARD ?= samr21-xpro
+include ../Makefile.tests_common
+
+USEMODULE += architecture
```

Comment:
This has no C files and it feels pretty fundamental - why not just put it in `core/include`?

Reply:
I'd like to keep `core` lean. If something is in `core`, PRs need two ACKs and users expect APIs to almost never change. (Of course we would avoid breaking API changes everywhere, but with core APIs it is extra hard.)